assets/js/modules/analytics-4/components/widgets/TopCitiesWidget.js

```diff
@@ -86,6 +86,17 @@ function TopCitiesWidget( { Widget } ) {
 
 	const { rows = [], totals = [] } = topCitiesReport || {};
 
+	const filteredCities = ( cityRows ) =>
+		cityRows.some(
+			( { dimensionValues } ) =>
+				dimensionValues[ 0 ].value === '(not set)'
+		)
+			? cityRows.filter(
+					( { dimensionValues } ) =>
+						dimensionValues[ 0 ].value !== '(not set)'
+			  )
+			: cityRows.slice( 0, 3 );
```

Comment:
As per this comment, https://github.com/google/site-kit-wp/issues/7884#issuecomment-2548444615, we should create a reusable function for this logic, please can you lift this out as a common helper and reuse it in `TopCountriesWidget`.

We can also simplify it a bit:


```suggestion
	const filteredCities = ( cityRows ) => {
		const setCities = cityRows.filter(
			( { dimensionValues } ) =>
				dimensionValues[ 0 ].value !== '(not set)'
		);
		
		// If there are still 4 rows in the list, we only want the top 3:
		return setCities.slice( 0, 3 );
	};
```